roles/os_firewall/vars/main.yml

```diff
@@ -0,0 +1,4 @@
+---
+os_firewall_pkgs: "{{ ['firewalld'] if os_firewall_use_firewalld else ['iptables', 'iptables-services'] }}"
+os_firewall_svc: "{{ 'firewalld' if os_firewall_use_firewalld else 'iptables' }}"
+os_firewall_mask_svcs: "{{ ['iptables'] if os_firewall_use_firewalld else ['firewalld'] }}"
```

Comment:
So I see that here you're trying to abstract out the logic so that we can have 1 yum command and it'll install either iptables or firewalld packages.

While I commend you for trying to keep with the DRY principle (which I subscribe too as well), what I prefer in this situation is that firewall/firewalld.yml have the yum calls for firewalld specifically, and that firewall/iptables.yml have the yum calls for iptables.

Same for the services, masking, etc.

It makes the logic much easier to follow, and any errors will be much clearer. Otherwise I have to jump through multiple files to follow the logic of how stuff got set to each value.

I don't believe this violates DRY, as I see these lines the same as method calls, whereas DRY is specifically for not defining multiple methods with the same business logic.

Also, I prefer to not do too much logic in jinja as the errors aren't as clearly laid out as they are in ansible, and they're harder to debug.

In other words, I prefer stuff like this (very simple to follow and debug):
```
# start of firewall/firewalld.yml
- name: Install firewalld packages
  yum: name={{ item }} state=present
  with_items:
    - firewalld

# start of firewall/iptables.yml
- name: Install iptables packages
  yum: name={{ item }} state=present
  with_items:
    - iptables
    - iptables-services
```